synapse/storage/devices.py

```diff
@@ -210,12 +224,15 @@ def _get_device_update_edus_by_remote(self, destination, from_stream_id, query_m
                 destination, user_id, from_stream_id
             )
             for device_id, device in iteritems(user_devices):
-                stream_id = query_map[(user_id, device_id)]
+                stream_id = query_map[(user_id, device_id)][0]
```

Comment:
according to the docstring, `query_map` is a dict to an int. You can't `[]-index` an int. I guess the docstring is lying.

Also I suggest you write:

```
stream_id, context = query_map[(user_id, device_id)]
```